pygfx/wgpu-py
# update to wgpu-native 22.1

I am really excited for better error handling, ~~compilation info~~(not yet implemented in wgpu-native, I might try to do that myself to move it along) and glsl const built-ins, so I already started this. 
Feel free to cherry pick my changes or commit into this branch if it helps.

I got a lot of time to work on this, but need to read up on error handling a bit.


(from the issue template:)
## Preparations

* [x] Perhaps warp up outstanding PR's.
* [x] Might be a good time to roll a release of wgpu-py.
* [x] Perhaps for pygfx too.

## Upstream
pending gfx-rs/wgpu-native#402
* [x] wgpu-native is updated to a recent wgpu-core.
* [x] wgpu-native uses a recent `webgpu.h`.
* [x] wgpu-native has a release with these changes.

## Update to latest IDL

*Optional, can also only update wgpu-native.*

*The lines below can be copied in the PR's top post.*

* [x] Download the latest `webgpu.idl` and place in the resources folder.
* [x] Run `python codegen` to apply the automatic patches to the code.
* [?] It may be necessary to tweak the `idlparser.py` to adjust to new formatting.
* [x] Check the diff of `flags.py`, `enums.py`, `structs.py` for any changes that might need manual work.
 * [x] Go through all FIXME comments that were added in `_classes.py`:
        * Apply any necessary changes.
        * Remove the FIXME comment if no further action is needed, or turn into a TODO for later.
        * Note that all new classes/methods/properties (instead those marked as hidden) need a docstring.
* [x] Run `python codegen` again to validate that all is well. Repeat the step above if necessary.
* [x] Make sure that the tests run and provide full coverage.
* [x] Make sure that the examples all work.
* [x] Make sure that pygfx works (again).
* [x] Update release notes for API changes.

## Update to latest wgpu-native

*Optional, can also only update IDL*

*The lines below can be copied in the PR's top post.*

* [x] Run `python download-wgpu-native.py --version xx` to download the latest webgpu.h and DLL.
* [x] Run `python codegen` to apply the automatic patches to the code.
* [?] It may be necessary to tweak the `hparser.py` to adjust to new formatting.
* [x] Diff the report for new differences to take into account.
* [x] Diff `wgpu_native/_api.py` to get an idea of what structs and functions have changed.
* [x] Go through all FIXME comments that were added in `_api.py`:
        * Apply any necessary changes.
        * Remove the FIXME comment if no further action is needed, or turn into a TODO for later.
* [x] Run `python codegen` again to validate that all is well. Repeat the steps above if necessary.
* [x] Make sure that the tests run and provide full coverage.
* [x] Make sure that the examples all work.
* [x] Make sure that pygfx works (again).
* [x] Update release notes for changes in behavior.

## Wrapping up

* [x] Update pygfx.
* [ ] Release new wgpu-py.
* [ ] Release new pygfx.
* [ ] This can be a good moment to deal with certain issues, or perhaps some issues can be closed.
